submariner/cable-driver-policy.md

```diff
@@ -0,0 +1,250 @@
+# Cable Driver Policy Support
+
+## Summary
+
+Submariner currently offers three cable drivers: VXLAN, IPSec or wireguard. However only one of these cable drivers
+is enabled across all the clusters in a ClusterSet; That is all clusters that join a broker must use the same cable
+driver. The goal of this proposal is to extend the current model to support the selection of different cable types
+between connected clusters. For example, an IPSec cable between public and private clouds but a VXLAN cable between
+the clusters in the same cloud.
+
+There are use cases that involve supporting a mix of cable drivers or cable driver configurations within the same
+ClusterSet. For example, a ClusterSet could contain three clusters, two that are on-premise and the other
+in a public cloud. Connections to the cloud cluster will need to use encryption while the on-premise clusters can use
+unencrypted connections, as they’re connected by a private link, to avoid the performance overhead of encryption.
+
+## Proposal
+
+This enhancement proposes a new policy to allow users to selectively deploy different cable types between clusters
+in a ClusterSet. The proposal assumes that connection between two clusters is symmetrical. If Cluster A is connected
+to Cluster B via a VXLAN cable type - then Cluster B is connected to Cluster A via a VXLAN cable type and not another
+cable type.
+
+This enhancement will propose a single API for cable driver policy support using custom resource definition (CRDs).
+
+### Cable driver policy API
+
+This following CRD is proposed:
+
+```Go
+type CablePolicies struct {
+    metav1.TypeMeta `json:",inline"`
+
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    ConnectionPolicies []CablePolicySpec `json:"spec"`
+}
+
+type CablePolicySpec struct {
+    // LeftClusterSelector identifies the Cluster resources representing the clusters on one end of a connection. An empty
+    // selector indicates wildcard, ie matches any cluster.
+    LeftClusterSelector metav1.LabelSelector `json:"leftClusterSelector,omitempty"`
+
+    // RightClusterSelector identifies the Cluster resources representing the clusters on the other end of a connection.
+    // An empty selector indicates wildcard, ie matches any cluster.
+    RightClusterSelector metav1.LabelSelector `json:"rightClusterSelector,omitempty"`
+
+    // Name of the cable driver implementation to use for this connection.
+    CableDriver string `json:"name"`
+
+    // Options specifies a ConfigMap to use for additional configuration parameters for the cable driver.
+    CableOptions *v1.ConfigMap `json:"options,omitempty"`
```

Comment:
I don't think we need to/should embed an entire `ConfigMap` CR. Wouldn't just `map[string]string` suffice here?

```suggestion
    CableDriverOptions *v1.ConfigMap `json:"cableDriverOptions,omitempty"`
```